connectors/src/connectors/google_drive/temporal/spreadsheets.ts

```diff
@@ -387,9 +387,24 @@ export async function syncSpreadSheet(
   );
 
   const sheetsAPI = google.sheets({ version: "v4", auth: oauth2client });
-  const spreadsheet = await sheetsAPI.spreadsheets.get({
-    spreadsheetId: file.id,
-  });
+  const spreadsheet = await (async () => {
```

Comment:
Why do we need an IIFE here?

Reply:
to keep the `const` for spreadsheet
can do a let above the try / catch if you prefer, tell me :+1:

Reply:
but then likely to have a dummy typeguard

Reply:
You don't need the typeguard since you will be returning in the catch 🙃. The best way would probably be to move this to its own function so we can keep the const and avoid the IIFE.